src/axes/axis.ts

```diff
@@ -104,6 +104,33 @@ export class Axis<D> extends Component {
     this._scale.offUpdate(this._rescaleCallback);
   }
 
+  /**
+   * Gets the tick label data at a particular point.
+   *
+   * @param {Point} point
+   */
+  public tickLabelAt(point: Point): D {
+    // only measure in the direction of the axis.
+    const pointValue = this.isHorizontal() ? point.x : point.y;
+    const tickValues = this._getTickValues();
+
+    const axisScale = this._scale;
+    let tickLabel: D;
+    this._tickLabelContainer.selectAll<SVGGElement, any>("." + Axis.TICK_LABEL_CLASS)
+      .each(function (data: D, index: number) {
+        const labelWidth = this.getBBox().width;
```

Comment:
this should be `height` for vertical axes right? Also, `getBBox()` is an expensive operation (e.g. on the order of milliseconds sometimes) so we should avoid it if possible